creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.js

```diff
@@ -1,23 +1,550 @@
+ * @param {Set<string>} param.unhealthyReplicasSet the endpoints of the user's replica set that are currently unhealthy
+ * @param {Object} param.replicaSetNodesToUserClockStatusesMap map of secondary endpoint strings to (map of user wallet strings to clock value of secondary for user)
+ * @param {string[]} param.enabledReconfigModes array of which reconfig modes are enabled
  */
 module.exports = async function (
-  jobId,
-  users, // TODO: This will be updated to only run for one user
-  unhealthyPeers,
-  userSecondarySyncMetricsMap,
-  replicaSetNodesToUserWalletsMap,
-  replicaSetNodesToUserClockStatusesMap
+  logger,
+  wallet,
+  userId,
+  primary,
+  secondary1,
+  secondary2,
+  unhealthyReplicasSet,
+  replicaSetNodesToUserClockStatusesMap,
+  enabledReconfigModes
 ) {
-  // TODO: Move snapback's `_aggregateOps` (decouple from the sync part of this function), `autoSelectCreatorNodes`, `determineNewReplicaSet`, and `issueUpdateReplicaSetOp` steps
-  // into the *monitoring queue* and then make them output a single reconfig to execute here
+  _validateJobData(
+    logger,
+    primary,
+    secondary1,
+    secondary2,
+    wallet,
+    unhealthyReplicasSet,
+    healthyNodes,
+    replicaSetNodesToUserClockStatusesMap,
+    enabledReconfigModes
+  )
+
+  /**
+   * Fetch all the healthy nodes while disabling sync checks to select nodes for new replica set
+   * Note: sync checks are disabled because there should not be any syncs occurring for a particular user
+   * on a new replica set. Also, the sync check logic is coupled with a user state on the userStateManager.
+   * There will be an explicit clock value check on the newly selected replica set nodes instead.
+   */
+  const { services: healthyServicesMap } = {}
+  await QueueInterfacer.getAudiusLibs().ServiceProvider.autoSelectCreatorNodes({
+    performSyncCheck: false,
+    whitelist: reconfigNodeWhitelist,
+    log: true
+  })
 
-  // TODO: Return data about updated replica set
+  const healthyNodes = Object.keys(healthyServicesMap)
+  if (healthyNodes.length === 0)
+    throw new Error(
+      'Auto-selecting Content Nodes returned an empty list of healthy nodes.'
+    )
+
+  try {
+    const newReplicaSet = await determineNewReplicaSet({
+      wallet,
+      primary,
+      secondary1,
+      secondary2,
+      unhealthyReplicasSet,
+      healthyNodes,
+      replicaSetNodesToUserClockStatusesMap,
+      enabledReconfigModes
+    })
+    const { errorMsg, issuedReconfig } = await issueUpdateReplicaSetOp(
+      userId,
+      wallet,
+      primary,
+      secondary1,
+      secondary2,
+      newReplicaSet,
+      logger
+    )
+  } catch (e) {
+    logger.error(
+      `ERROR issuing update replica set op: userId=${userId} wallet=${wallet} old replica set=[${primary},${secondary1},${secondary2}] | Error: ${e.toString()}`
+    )
+  }
+
+  // TODO: Return data about updated replica set or failure
```

Comment:
will do this and tests later. just getting consensus on the rest of the PR before making it even bigger since replica set updates take more work to simulate locally